tests/python_package_test/test_basic.py

```diff
@@ -621,3 +621,23 @@ def test_no_copy_when_single_float_dtype_dataframe(dtype):
     built_data = lgb.basic._data_from_pandas(df, feature_name, None, None)[0]
     assert built_data.dtype == dtype
     assert np.shares_memory(X, built_data)
+
+
+def test_feature_num_bin():
```

Comment:
Thanks a lot for working on this!

What about adding one more test that involves [max_bin_by_feature](https://lightgbm.readthedocs.io/en/latest/Parameters.html#max_bin_by_feature) parameter?

Reply:
Added in 9a97e90.